WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -124,9 +127,9 @@ private void StepOutputRegistrationPhase()
 
 					// If timeout we must fill up the outputs to build a reasonable transaction.
 					// This won't be signed by the alice who failed to provide output, so we know who to ban.
-					if (diff > coinjoin.Parameters.AllowedOutputAmounts.Min)
+					var diffMoney = Money.Satoshis(diff) - coinjoin.Parameters.FeeRate.GetFee(Config.BlameScript.EstimateOutputVsize());
+					if (diffMoney > coinjoin.Parameters.AllowedOutputAmounts.Min)
```

Comment:
I think this is more correct behaviour:

```suggestion
					if (!allReady && diffMoney > coinjoin.Parameters.AllowedOutputAmounts.Min)
```

because if all alices agreed to sign, then whatever amount is left over, they consented to use as extra mining fees.

alternatively it can be something like:

```suggestion
					if (diffMoney > alices.Count() * x)
```

where `x` is some reasonable value for a maximum reasonable mining fee (designed to allow recovery of funds in the case of a client bug, or unsupported concurrent use of the same seed in multiple clients)

Reply:
Added.